src/components/MultiGestureCanvas/usePanGesture.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-param-reassign */
+import {Gesture} from 'react-native-gesture-handler';
+import {useDerivedValue, useSharedValue, withDecay, withSpring} from 'react-native-reanimated';
+import * as MultiGestureCanvasUtils from './utils';
+
+const PAN_DECAY_DECELARATION = 0.9915;
+
+const usePanGesture = ({canvasSize, contentSize, zoomScale, totalScale, offsetX, offsetY, panTranslateX, panTranslateY, isSwipingInPager, stopAnimation}) => {
+    // The content size after fitting it to the canvas and zooming
+    const zoomedContentWidth = useDerivedValue(() => contentSize.width * totalScale.value, [contentSize.width]);
+    const zoomedContentHeight = useDerivedValue(() => contentSize.height * totalScale.value, [contentSize.height]);
+
+    // Pan velocity to calculate the decay
```

Comment:
What is "decay"?

Reply:
It's a type of animation that emulates the physics of a gesture after the finger's have been released. E.g. it continues panning after the pan gesture ends, so that it doesn't cut off immediately.

I'll further improve the comment by also linking to the REA docs https://docs.swmansion.com/react-native-reanimated/docs/animations/withDecay/